butterknife-compiler/src/main/java/butterknife/compiler/ButterKnifeProcessor.java

```diff
@@ -247,44 +253,75 @@
       }
     }
 
-    // Process each @Unbinder element.
-    for (Element element : env.getElementsAnnotatedWith(Unbinder.class)) {
-      if (!SuperficialValidation.validateElement(element)) continue;
-      try {
-        parseBindUnbinder(element, targetClassMap, erasedTargetNames);
-      } catch (Exception e) {
-        logParsingError(element, Unbinder.class, e);
-      }
-    }
-
     // Try to find a parent binder for each.
+    HashSet<BindingClass> topLevelClasses = new HashSet<>();
     for (Map.Entry<TypeElement, BindingClass> entry : targetClassMap.entrySet()) {
       TypeElement parentType = findParentType(entry.getKey(), erasedTargetNames);
       if (parentType != null) {
-        String parentClassFqcn = getFqcn(parentType);
         BindingClass bindingClass = entry.getValue();
-        bindingClass.setParentViewBinder(parentClassFqcn + BINDING_CLASS_SUFFIX);
-        // Check if parent requested an unbinder.
         BindingClass parentBindingClass = targetClassMap.get(parentType);
-        if (parentBindingClass.hasUnbinder()) {
-          // Even if the child doesn't request an unbinder explicitly, we need to generate one.
-          if (!bindingClass.hasUnbinder()) {
-            bindingClass.requiresUnbinder(null);
-          }
-          // Check if the parent has a parent unbinder.
-          if (parentBindingClass.getParentUnbinder() != null) {
-            bindingClass.setParentUnbinder(parentBindingClass.getParentUnbinder());
-          } else {
-            bindingClass.setParentUnbinder(parentClassFqcn + BINDING_CLASS_SUFFIX + "."
-                + UnbinderBinding.UNBINDER_SIMPLE_NAME);
-          }
-        }
+        bindingClass.setParent(parentBindingClass);
+        parentBindingClass.addDescendant(bindingClass);
+      } else {
+        // This is a top level parent
+        topLevelClasses.add(entry.getValue());
       }
     }
 
+    Observable.from(topLevelClasses)
+        .flatMap(topLevelClass -> {
+          if (topLevelClass.hasViewBindings()) {
+            // It has an unbinder class and it will also be the highest unbinder class for all
+            // descendants
+            topLevelClass.setHighestUnbinderClassName(topLevelClass.getUnbinderClassName());
+          } else {
+            // No unbinder class. Null it out so we know we can just return the NOP unbinder
```

Comment:
All your comments are missing periods. I'm commenting on the most problematic one because it contains a period for the first sentence but not the second. [Comments should be sentences](http://nedbatchelder.com/blog/201401/comments_should_be_sentences.html).